s2cloudless/cloud_detector.py

```diff
@@ -0,0 +1,162 @@
+    of shape n x m, where each pixel's value is bound between 0 (clear-sky-like pixel)
+    and 1 (cloud-like pixel).
+
+    User can control cloud probability threshold and/or post-processing steps -
+    convolution with disk (with user defined filter size) and dilation with disk
+    (with user defined filter size).
+
+    :param threshold: Cloud probability threshold. All pixels with cloud probability above
+                      threshold value are masked as cloudy pixels. Default is 0.4.
+    :type threshold: float
+    :param all_bands: Flag specifying that input images will consists of all 13 Sentinel-2 bands.
+    :type all_bands: bool
+    :param average_over: Size of the disk in pixels for performing convolution (averaging probability
+                         over pixels). Value 0 means do not perform this post-processing step.
+                         Default is 1.
+    :type average_over: int
+    :param dilation_size: Size of the disk in pixels for performing dilation. Value 0 means do not perform
+                          this post-processing step. Default is 1.
+    :type dilation_size: int
+    :param model_filename: Location of the serialised model. If None the default model provided with the
+                           package is loaded.
+    :type model_filename: str or None
+    """
+
+    def __init__(self, threshold=0.4, all_bands=False, average_over=1, dilation_size=1, model_filename=None):
+
+        self.threshold = threshold
+        self.all_bands = all_bands
+        self.average_over = average_over
+        self.dilation_size = dilation_size
+
+        if model_filename is None:
+            package_dir = os.path.dirname(__file__)
+            model_filename = os.path.join(package_dir, 'models', MODEL_FILENAME)
+        self.model_filename = model_filename
+
+        self._classifier = None
+
+        if average_over > 0:
+            self.conv_filter = disk(average_over) / np.sum(disk(average_over))
+
+        if dilation_size > 0:
+            self.dilation_filter = disk(dilation_size)
+
+    @property
+    def classifier(self):
+        """
+        Provides a classifier object. It also loads it if it hasn't been loaded yet. This way the classifier is loaded
+        only when it is actually required.
+        """
+        if self._classifier is None:
+            self._classifier = PixelClassifier(Booster(model_file=self.model_filename))
+
+        return self._classifier
+
+    def get_cloud_probability_maps(self, data, **kwargs):
+        """
+        Runs the cloud detection on the input images (dimension n_images x n x m x 10
+        or n_images x n x m x 13) and returns an array of cloud probability maps (dimension
+        n_images x n x m). Pixel values close to 0 indicate clear-sky-like pixels, while
+        values close to 1 indicate pixels covered with clouds.
+
+        :param data: input Sentinel-2 image obtained with Sentinel-Hub's WMS/WCS request
```

Comment:
Fixed. It actually shouldn't say anything where this data should be obtained from. `S2PixelCloudDetector` is not tied to a specific input source. So the data can be from SH but it can also be from somewhere else. Only `CloudMaskReqeust` class is the one that relies on SH service.